ai-services/cmd/ai-services/cmd/application/create.go

```diff
@@ -160,6 +161,54 @@ var createCmd = &cobra.Command{
 			}
 		}
 
+		if !skipImageDownload {
+			// Download container images if flag is set to false (default: false)
+			logger.Infoln("Downloading container images as part of application creation...")
+			images, err := helpers.ListImages(templateName, appName)
+			if err != nil {
+				return fmt.Errorf("failed to list container images: %w", err)
+			}
+			logger.Infoln("Downloading container images required for application template " + templateName + ":")
+			for _, image := range images {
+				logger.Infoln("Downloading image: " + image + "...")
+				if err := utils.Retry(retryCount, retryInterval, nil, func() error {
+					return runtime.PullImage(image, nil)
+				}); err != nil {
+					return fmt.Errorf("failed to download image: %w", err)
+				}
+			}
+			logger.Infoln("Downloading container images completed.")
+		} else {
+			logger.Infoln("Skipping container image download as per the flag --skip-image-download=true")
+			// Verify that images exist locally
+			images, err := helpers.ListImages(templateName, appName)
+			if err != nil {
+				return fmt.Errorf("failed to list container images: %w", err)
+			}
+			lImages, err := runtime.ListImages()
+			if err != nil {
+				return fmt.Errorf("failed to list local images: %w", err)
+			}
+			notfoundImages := []string{}
+			for _, image := range images {
+				found := false
+				for _, lImage := range lImages {
+					// Check both RepoTags and RepoDigests for presence
+					if slices.Contains(lImage.RepoTags, image) || slices.Contains(lImage.RepoDigests, image) {
+						found = true
+						break
+					}
+				}
+				if !found {
+					notfoundImages = append(notfoundImages, image)
+				}
+			}
+			if len(notfoundImages) > 0 {
+				return fmt.Errorf("some required images are not present locally: %v. Either pull the image manually or rerun create command without --skip-image-download flag", notfoundImages)
```

Comment:
We can also show the locally not available images as well, useful for user to pull.